modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java

```diff
@@ -0,0 +1,343 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.google.common.collect.Lists;
+import com.smartbear.readyapi.client.model.ProjectResultReport;
+import com.smartbear.readyapi.client.model.TestCaseResultReport;
+import com.smartbear.readyapi.client.model.TestStepResultReport;
+import com.smartbear.readyapi.client.model.TestSuiteResultReport;
+import com.smartbear.readyapi4j.TestRecipe;
+import com.smartbear.readyapi4j.TestRecipeBuilder;
+import com.smartbear.readyapi4j.execution.Execution;
+import com.smartbear.readyapi4j.execution.RecipeExecutor;
+import com.smartbear.readyapi4j.facade.execution.RecipeExecutorBuilder;
+import com.smartbear.readyapi4j.testserver.execution.ProjectExecutionRequest;
+import com.smartbear.readyapi4j.testserver.execution.ProjectExecutor;
+import com.smartbear.readyapi4j.testserver.execution.TestServerClient;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.model.Resource;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.shared.filtering.MavenFilteringException;
+import org.apache.maven.shared.filtering.MavenResourcesExecution;
+import org.apache.maven.shared.filtering.MavenResourcesFiltering;
+import org.apache.maven.shared.model.fileset.FileSet;
+import org.apache.maven.shared.model.fileset.util.FileSetManager;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import static com.smartbear.readyapi4j.testserver.execution.ProjectExecutionRequest.Builder.forProjectFile;
+
+@Mojo(name = "run")
+public class RunMojo
+        extends AbstractMojo {
+    @Component
+    private MavenResourcesFiltering resourcesFiltering;
+
+    @Component
+    private MavenProject mavenProject;
+
+    @Component
+    private MavenSession mavenSession;
+
+    @Parameter
+    private Map properties;
+
+    @Parameter
+    private String environment;
+
+    @Parameter(defaultValue = "false")
+    private boolean async;
+
+    @Parameter
+    private String callback;
+
+    @Parameter
+    private boolean disableFiltering;
+
+    @Parameter(defaultValue = "true")
+    private boolean failOnFailures;
+
+    @Parameter(required = true, property = "ready-api-test-server.username")
```

Comment:
suggest we use the same name as the system-properties use in the facade;
- testserver.endpoint
- testserver.user
- testserver.password